move sorting algorithms into lists and memory topic (#18)
Ada-Developers-Academy/core-cs-fundamentals: lists-and-memory/sorting-algorithms-lists-memory.md

| Stable                                    | A sorting algorithm is _stable_ if it guarantees that the relative ordering of two values having the same sort key is the same after the sort as it was before the sort.            | "Since that sorting algorithm is stable, we can apply it multiple times to the same data to sort records based on multiple attributes!"                                                                                                  |

## Introduction

Imagine a stack of index cards, where each index card has the numbers 1 through 100. Imagine that these cards are shuffled in a random, unknowable order.

If you're given the task to sort these index cards in ascending order, what would you do?

Picture yourself. Do you scan through the pile over and over again, slowly changing the order until it's all sorted? Do you use a table and make two separate piles? Do you fan through the cards comparing each card to its neighbor in the pile?

If you take two people and ask them to sort the same pile of cards, they will likely take different approaches.

> Machines need to be told how to think through these questions too, but with fewer index cards and more data. The strategies we tell a computer to use for sorting a list of values are called _sorting algorithms_.

Reviewer: ```suggestion
Machines need to be told how to think through these questions too, but data instead of index cards. The strategies we tell a computer to use for sorting a list of values are called _sorting algorithms_.
```

Reviewer 2: +1

We saw the implementation of bubble sort in the previous lesson. Follow your curiosity to further review the details of [bubble sort](../sorting-algorithms/bubble-sort.md) and [selection sort](../sorting-algorithms/selection-sort.md). These Learn lessons are optional to review.

These O(n<sup>2</sup>) algorithms are fine when the array length _n_ is small, but quickly cease to be useful as the size of the array increases. Bubble sort and insertion sort can perform well in the best case, but they do not scale well.

They **do** have the advantage of being approachable for discussion.

Moving beyond these O(n<sup>2</sup>) algorithms, we will discuss _merge sort_ later in the curriculum after we cover _recursion_, a powerful algorithmic strategy. 

## Summary

There are many sorting algorithms that have been developed. We will discuss some of them in the coming lessons, but we can learn more about other sorting methods at [geeksforgeeks.org/sorting-algorithms/](http://www.geeksforgeeks.org/sorting-algorithms/).

> Not all sorting algorithms are created equally, so we will want to keep the performance characteristics of each algorithm in mind when selecting one to use.

Reviewer: ```suggestion
Not all sorting algorithms are created equal, so we will want to keep the performance characteristics of each algorithm in mind when selecting one to use.
```

Reviewer 2: +1
